knowledge-base/githubocto/repo-visualizer/action-security.yml

```diff
@@ -0,0 +1,14 @@
+name: 'Repo Visualizer'
+github-token:
+  action-input:
+    input: github_token
+    is-default: true
+  permissions:
+    contents: write
+    contents-if: ${{ !contains(with, 'should_push') || with['should_push'] == 'true' }}
+    contents-reason: to read and push into repo directories
```

Comment:
The reason should be `to commit and push diagram` 
Try to use the same/ similar words as written in the Action. 
https://github.com/githubocto/repo-visualizer/blob/ccc127b6bb1ef7123ea50544cc22a90e9f5f6f3d/src/index.jsx#L76

Also, please add link to reason as a comment